compact.go

```diff
@@ -244,7 +266,7 @@ func (c *compactor) write(dir string, blocks ...Block) (err error) {
 	if err := renameFile(tmp, dir); err != nil {
 		return errors.Wrap(err, "rename block dir")
 	}
-	for _, b := range blocks[1:] {
+	for _, b := range blocks {
```

Comment:
Before blocks [1,2,3,4] got compacted int [1,4] – so only 2 and 3 were deleted while 1 was overwriten with a block spanning a larger range.
As we just use ULIDs now, all blocks have to be removed.